nuskybgd/model.py

```diff
@@ -523,71 +523,138 @@ def applymodel_intbgd(presets, refspec, bgddetimsum, model_num, src_number=None,
                 attr = 'lorentz_%d' % (attr_n + 1)
-                m_line = getattr(m, attr)
-                m_line.LineE.values = mod_intbgd[attr][fpm]['linee']
-                m_line.Width.values = mod_intbgd[attr][fpm]['width']
-                m_line.LineE.frozen = True
-                m_line.Width.frozen = True
-
-                norm_preset = np.sum(
-                    bgddetimsum[spec_arrinx] * np.array(
+                line_norm = np.sum(
+                        bgddetimsum[spec_arrinx] * np.array(
                         mod_intbgd[attr][fpm]['ifactors'])
                 ) / np.sum(bgddetimsum[spec_arrinx])
-
-                if fix_line_ratios:
-                    lorentz_5_norm_npar = spec_arrinx * m.nParameters + 16
-
-                    m_line.norm.link = '%e * %s:p%d' % (
-                        norm_preset / m.lorentz_5.norm.values[0],
-                        model_name,
-                        lorentz_5_norm_npar)
+                
+                if not ref_set:
+                    ref_set = True
+                    ref_norm = line_norm
+                    ref_par = int(parnum+2)+ m_ref_npar_offset
+                    this_par = {parnum:f"{mod_intbgd[attr][fpm]['linee']}, -0.1",
+                            parnum+1:f"{mod_intbgd[attr][fpm]['width']}, -0.1",
+                            parnum+2:f"{line_norm}, 0.1"}
                 else:
-                    m_line.norm.values = norm_preset
+                    rel_norm = line_norm / ref_norm                
+                    this_par = {parnum:f"{mod_intbgd[attr][fpm]['linee']}, -0.1",
+                                parnum+1:f"{mod_intbgd[attr][fpm]['width']}, -0.1",
+                                parnum+2:f"={model_name}:{ref_par}*{rel_norm}"}
+                par_dict.update(this_par)
+                parnum += 3
+
+                continue
+                
+                # Replaced by above
+#                 attr = 'lorentz_%d' % (attr_n + 1)
+#                 m_line = getattr(m, attr)
+#                 m_line.LineE.values = mod_intbgd[attr][fpm]['linee']
+#                 m_line.Width.values = mod_intbgd[attr][fpm]['width']
+#                 m_line.LineE.frozen = True
+#                 m_line.Width.frozen = True
+# 
+#                 norm_preset = np.sum(
+#                     bgddetimsum[spec_arrinx] * np.array(
+#                         mod_intbgd[attr][fpm]['ifactors'])
+#                 ) / np.sum(bgddetimsum[spec_arrinx])
+# 
+#                 if fix_line_ratios:
+#                     lorentz_5_norm_npar = spec_arrinx * m.nParameters + 16
+# 
+#                     m_line.norm.link = '%e * %s:p%d' % (
+#                         norm_preset / m.lorentz_5.norm.values[0],
+#                         model_name,
+#                         lorentz_5_norm_npar)
+#                 else:
+#                     m_line.norm.values = norm_preset
+
+            # Actually apply the changes here
+            m.setPars(par_dict)
+
```

Comment:
(Note that this part concerns the line norms in the reference spectra)

I added this fix_line_ratios switch to toggle whether ref spectra lines from the 6th line onwards are linked to the 5th one, in response to Dan's request to change the default behavior to *not* tie them together. The ref_set switch here would undo that :(